### DISCUSSION TITLE
A Vue version of the `QueryClientProvider` component

### DISCUSSION THREAD

**Discussion by ElisePatrikainen** [TanStack/query#4565] [Ideas] [answered]
It seems the `QueryClientProvider` component is a React component, and therefore can't be used in a Vue application. It would be great to have a Vue equivalent. 

Thanks!

**Comment by DamianOsipiuk** [comment, marked as answer]
Hi,
Vue has a different convention of setting up globally accessible properties. You can check `basic` example to see how it works - https://github.com/TanStack/query/blob/main/examples/vue/basic/src/main.ts

**Reply by DamianOsipiuk** [reply]
```ts
const vueQueryPluginOptions: VueQueryPluginOptions = {
  queryClientConfig: {
    defaultOptions: { queries: { staleTime: 3600 } },
  },
};
app.use(VueQueryPlugin, vueQueryPluginOptions);
```

If you need more docs with snippets, check this page, before it's migrated to tanstack docs - https://vue-query-next.vercel.app/#/guides/custom-client

**Reply by ElisePatrikainenIAD** [reply]
Thank you very much for your quick reply, this solves my issue perfectly well! 🙏
